documentation/docs/internals/02-architecture.md

```diff
@@ -0,0 +1,10 @@
+# Architecture
+
+Fuego's architecture rely on the following components:
+
+- **Engine**: The engine is responsible for handling the request and response. It is the core of Fuego.
```

Comment:
Should we mention this is where the OpenAPI/OpenAPIConfig is encapsulated?

Reply:
Oh yes why not! We'll definitely revisit this page haha, it's far from complete.